### PR TITLE
Support retrieval of a list of files with specific extension from specific path

### PR DESCRIPTION
**Pull request by AlexeiVainshtein**

### CODE REVIEW

File: utils/io/fileutils/files.go [hunk 1]
```diff
@@ -133,6 +133,32 @@ func ListFilesRecursiveWalkIntoDirSymlink(path string, walkIntoDirSymlink bool)
 	return
 }
 
+// Return the list of files in the specified path that has the extension
+func ListFilesWithSpecificExtension(path, ext string) ([]string, error) {
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: I suggest shortening the function name to *ListFilesWithExtension*.

File: utils/io/fileutils/files.go [hunk 2]
```diff
@@ -133,6 +133,32 @@ func ListFilesRecursiveWalkIntoDirSymlink(path string, walkIntoDirSymlink bool)
 	return
 }
 
+// Return the list of files in the specified path that has the extension
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: // Return the list of files in the specified path that has the extension -->
// Return all files with the specified extension in the specified path. Not recursive.

File: utils/io/fileutils/files.go [hunk 3]
```diff
@@ -133,6 +133,32 @@ func ListFilesRecursiveWalkIntoDirSymlink(path string, walkIntoDirSymlink bool)
...
+func ListFilesWithSpecificExtension(path, ext string) ([]string, error) {
+	sep := GetFileSeparator()
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: I think that the *GetFileSeparator()* function shoukd be removed from thr code base and replaced with ```os.PathSeparator```

File: utils/io/fileutils/files.go [hunk 4]
```diff
@@ -133,6 +133,32 @@ func ListFilesRecursiveWalkIntoDirSymlink(path string, walkIntoDirSymlink bool)
...
+	for _, f := range files {
+		if !strings.HasSuffix(f.Name(), ext) {
+			continue
+		}
+		filePath := path + f.Name()
+		exists, err := IsFileExists(filePath, false)
+		if err != nil {
+			return nil, err
+		}
+		if exists || IsPathSymlink(filePath) {
```
Comment (eyalbe4) [inline utils/io/fileutils/files.go]: I belive that you are interested in finding only files or symlinks to files.
I'm not 100% sure, but there's a chance that the above line will also find symlinks to directories. Worth checking just to be sure.

File: utils/io/fileutils/files.go (hunk 1 above)
Comment (AlexeiVainshtein) [reply utils/io/fileutils/files.go]: Done

File: utils/io/fileutils/files.go (hunk 2 above)
Comment (AlexeiVainshtein) [reply utils/io/fileutils/files.go]: Done

File: utils/io/fileutils/files.go (hunk 3 above)
Comment (AlexeiVainshtein) [reply utils/io/fileutils/files.go]: Per our discussion, I'm keeping the function and just replacing with string(os.PathSeparator)

File: utils/io/fileutils/files.go (hunk 4 above)
Comment (AlexeiVainshtein) [reply utils/io/fileutils/files.go]: I've changed it to ignore directories.

**Pull request merged**